Add charset to imap component

ZiroNL opened the pull request:
## Description:
The imap component did not work with office 365 because of the wrong charset.
The default charset is UTF-8 in the imap component, Office 365 does not support this.

(This is my first PR, so a little help is appreciated :) )

## Example entry for `configuration.yaml`:
```yaml
  - platform: imap
    server: outlook.office365.com
    port: 993
    username: email@address.com
    password: password
    charset: US-ASCII
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

PR #10967


[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

ZiroNL committed:
Adding charset to imap component

homeassistant commented:
Hi @ZiroNL, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#28154).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob committed:
Update translations

balloob committed:
Version bump to 102.0.dev0"

rytilahti committed:
Bump songpal to fix a regression (#28115)

The new release fixes a single regression from requests to aiohttp conversion.
Some devices do not respond with the correct mimetype which was not enforced
by requests but is enforced by aiohttp.

Related PR https://github.com/rytilahti/python-songpal/pull/59

Commit:
Cover all possible values for venstar operation_mode (#27949)

* cover all possible values for operation_mode

* Update climate.py

* Update climate.py

* Update climate.py

mapped homeassistant constants to client ones so we don't have to check for both

* black + pylint

* move mode_map to __init__()

* Update climate.py

* Update climate.py

homeassistant committed:
[ci skip] Translation update

djpremier committed:
Add onvif local datetime support (#26656)

* Update camera.py

* Add onvif local datetime support

* Correct capture TimeZone

* Replace one line if-statement by if-block

kennedyshead committed:
Fix Venstar formatting to restore clean CI (#28171)

florisvdk committed:
Add Unifi Led (#27475)

* Added Unifi Led

* fixed manifest

* fixed style issue

* removed unused setting

* added sugested changes.

* fixed order

* fixed settings that are required

* Fix review issues

* fix variable name that was too short

* Testing something

* Reverted to a previous version for testing

* Reverted testing changes.

Adminiuga committed:
Fixes/zha ieee tail (#28160)

* Fix ZHA entity_id assignment.

* Update tests.

scop committed:
Modernize Huawei LTE (#26675)

* Modernization rework

- config entry support, with override support from huawei_lte platform in YAML
- device tracker entity registry support
- refactor for easier addition of more features
- internal code cleanups

* Remove log level dependent subscription/data debug hack

No longer needed, because pretty much all keys from supported
categories are exposed as sensors.

Closes https://github.com/home-assistant/home-assistant/issues/23819

* Upgrade huawei-lte-api to 1.4.1

https://github.com/Salamek/huawei-lte-api/releases

* Add support for access without username and password

* Use subclass init instead of config_entries.HANDLERS

* Update huawei-lte-api to 1.4.3 (#27269)

* Convert device state attributes to snake_case

* Simplify scanner entity initialization

* Remove not needed hass reference from Router

* Return explicit None from unsupported old device tracker setup

* Mark unknown connection errors during config as such

* Drop some dead config flow code

* Run config flow sync I/O in executor

* Parametrize config flow login error tests

* Forward entry unload to platforms

* Async/sync fixups

* Improve data subscription debug logging

* Implement on the fly add of new and tracking of seen device tracker entities

* Handle device tracker entry unload cleanup in component

* Remove unnecessary _async_setup_lte, just have code in async_setup_entry

* Remove time tracker on unload

* Fix to not use same mutable default subscription set for all routers

* Pylint fixes

* Remove some redundant defensive device tracker code

* Add back explicit get_scanner None return, hush pylint

* Adjust approach to set system_options on entry create

* Enable some sensors on first add instead of disabling everything

* Fix SMS notification recipients default value

* Add option to skip new device tracker entities

* Fix SMS notification recipient option default

* Work around https://github.com/PyCQA/pylint/issues/3202

* Remove unrelated type hint additions

* Change async_add_new_entities to a regular function

* Remove option to disable polling for new device tracker entries

vangorra committed:
Use latest withings_api module (#27817)

* Using latest winthings_api module.
Drastically reduced complexity of tests.

* Removing import source.

* Fixing test requirements.

* Using requests_mock instead of responses module.

* Updating file formatting.

* Removing unused method.

* Adding support for new OAuth2 config flow.

* Addressing PR feedback.
Removing unecessary base_url from config, this is a potential breaking change.

* Addressing PR feedback.

OttoWinter committed:
Bump aioesphomeapi to 2.4.1 (#28170)

* Bump aioesphomeapi to 2.4.1

* Update requirements

* Bump to 2.4.2

ludeeus committed:
Tradfri config flow enhancements (#28179)

OttoWinter committed:
Fix ESPHome stacktraces when removing entity and shutting down (#28185)

ludeeus committed:
Add stop feature to tradfri covers (#28180)

* Tradfri cover enhancements

* tradfri dependency update

* Revert addition of battery attrubite

* Remove the supported_features property

* Remove unwanted file

djpremier committed:
Fix tzinfo type for onvif component (#28178)

kennedyshead committed:
Adding device_class to samsungtv (#28168)

* Adding device_id to samsungtv

* Lint

* Adding test

BKPepe committed:
Bump python-slugify to 4.0.0 (#28186)

ludeeus committed:
Removes unwanted tradfri battery sensor (#28181)

Commit:
Partially revert tensorflow import move (#28184)

* Revert "Refactor imports for tensorflow (#27617)"

This reverts commit 5a83a92390e8a3255885198c80622556f886b9b3.

* move only some imports to top

* fix lint

* add comments

scop committed:
Huawei LTE sensor metadata update (#28187)

homeassistant committed:
[ci skip] Translation update

escoand committed:
Add Samsung TV automatic protocol detection (#27492)

* added automatic protocol detection

* fix logger tests

* fix async tests

* add missin const.py

* fix log formatting

* wait for first update call

* migrate first tests

* migrated all test functions

* started to use state machine

* updated all tests to use async_setup_component

* slove hints

* update tests

* get state at correct position

* remove impossible tests

* fix autodetect tests

* use caplog fixture

* add test for duplicate

* catch concrete exceptions

* don't mock samsungctl exceptions

* add test for discovery

* get state when possible

* add test for autodetect without connection

guillempages committed:
[homematic]Pass channel to light color functions (#27306)

The device HmIP-BSL has two independent LEDs on two different channels,
so the channel needs to be explictly specified when setting the color.

guillempages committed:
[homematic]Add support for HmIP-BSL LEDs (#27307)

With this commit, 3 entities are created for the HmIP-BSL device:
2 lights for the two independent LEDs and 1 switch for the relais

gngj committed:
move hass-frontend import back down (#28203)

jjlawren committed:
Update Plex via websockets (#28158)

* Save client identifier from auth for future use

* Use websocket events to update Plex

* Handle websocket disconnections

* Use aiohttp, shut down socket cleanly

* Bad rebase fix

* Don't connect websocket during config_flow validation, fix tests

* Move websocket handling to external library

* Close websocket session on HA stop

* Use external library, revert unnecessary test change

* Async & lint fixes

* Clean up websocket stopper on entry unload

* Setup websocket in component, pass actual needed object to library

OnFreund committed:
Address post-merge coolmaster config flow code review (#28163)

* Address post-merge code review comments

* Use component path for 3rd party lib

ZiroNL commented:
Why did the checks failed?

scop committed:
Add presentation URL to SSDP discovery info (#28196)

bramkragten committed:
Updated frontend to 20191025.0 (#28208)

ochlocracy committed:
Change Alexa default display category based on media_player device_class (#28191)

* Support default display category based one media_player device_class.

* Support default display category based one media_player device_class.

shmooey committed:
Add device condition support for media_player (#28161)

* Add device condition for media_player

* Fix typo in strings

gngj committed:
Fix microsoft tts (#28199)

* Update pycsspeechtts

From 1.0.2 to 1.0.3 as the old one is using an api that doesn't work

* Give a option to choose region

Api is now region dependent, so gave it a config

shmooey committed:
Add device condition support for device_tracker (#28190)

Poeschl committed:
Increased python-eq3bt version to latest (0.1.11) (#28175)

ochlocracy committed:
Add support for supportedOperations to Alexa.PlaybackController (#28212)

* Added support for supportedOperations to Alexa.PlaybackController

* Added support for supportedOperations to Alexa.PlaybackController

emontnemery committed:
Fix broken deconz trigger (#28211)

ochlocracy committed:
Improved Alexa ThermostatController thermostatMode handling (#28176)

* Update ThermostatController to map directives to supported modes and add support for CUSTOM mode.

* Removed erroneous config value from test.

* Removed unnecessary use of a comprehension by dumbing down comment so pylint could comprehend.

* Removed erroneous import variable caused by removing erroneous config value from test.

* Removed unnecessary use of a comprehension.

* Reverted Removal or erroneous import variable and erroneous config value from test.
Apparently need for additional tests outside this component. Whoops.

emontnemery committed:
Add additional device conditions to cover (#27830)

* Add additional device conditions to cover

* Add default value

* Add test

* Use numeric_state condition instead of template condition

ochlocracy committed:
Implement Alexa.PlaybackStateReporter Interface for alexa (#28215)

emontnemery committed:
Add above and below to sensor condition extra_fields (#27364)

* Add above and below to sensor condition extra_fields

* Change unit_of_measurement to suffix in extra_fields

* Check if sensor has unit when getting capabilities

* Improve tests

balloob committed:
Add cloud account linking support (#28210)

* Add cloud account linking support

* Update account_link.py

homeassistant committed:
[ci skip] Translation update

scop committed:
Add Huawei LTE mobile data switch support (#28188)

* Add Huawei LTE mobile data switch support

* Remove stale comment

* Do HA state updates in base entity

fabaff committed:
Upgrade speedtest-cli to 2.1.2 (#28216)

frenck commented:
It fails the formatting check @ZiroNL. We use Black to format our code.
Please run `black --fast homeassistant` from the project root.

homeassistant committed:
[ci skip] Translation update

Commit:
Add available state to unifiled integration (#28189)

* Added Unifi Led

* fixed manifest

* fixed style issue

* removed unused setting

* added sugested changes.

* fixed order

* fixed settings that are required

* Fix review issues

* fix variable name that was too short

* Testing something

* Reverted to a previous version for testing

* Reverted testing changes.

* Add available status and increase version of unifiled package version.

* No io in init function.

ZiroNL committed:
Used black to format code

ZiroNL committed:
Merge error

ZiroNL committed:
Used black to format code.

ZiroNL commented:
Uhh, what happend? :) Closing this one!
